Sources/main.cpp

```diff
@@ -99,6 +101,12 @@ static int PlatformMain(std::string const &host, std::string const &port) {
 
   std::unique_ptr<Socket> serverSocket = CreateSocket(host, port, false);
 
+#if defined(OS_POSIX)
```

Comment:
I prefer this as well. Perhaps you can put an OS ifdef at the beginning of daemonize() that kills ds2 if it detects non-posix? That would be one OS ifdef, instead of the three you have in main.cpp.